### PR TITLE
Convert state-machine/ext traces to use `tracing`

### PR DESCRIPTION
**Pull request by insipx**
Rel #7780 

Currently all traces in `state-machine` evaluate to noop when compiled to a WASM binary. When tracing historical runtimes, traces that hint at storage commits, transactions, and gets are important to problems such as balance reconciliation. I figured the best compromise was converting traces in `state-machine` to use `sp_tracing`. This way, if tracing is not needed these messages will still be noops when compiled without `with-tracing`, but can be enabled otherwise.

This PR is in draft since it would be possible to convert all of `state-machine` to use `tracing` spans/events, however this PR  only converts ext.rs. If it is desirable to convert the rest of `state-machine` and remove the macros that evaluate to noops, then I can add more to the PR. It might be good to do that for consistency reasons.

I'm also unsure of the workflow the previous `log` traces were used in before, so confirmation that `sp_tracing` accomplishes the same thing would be good

### CODE REVIEW

File: primitives/state-machine/src/ext.rs [hunk 1]
```diff
@@ -589,26 +610,32 @@ where
...
 	fn storage_start_transaction(&mut self) {
-		self.overlay.start_transaction()
+		self.overlay.start_transaction();
+		sp_tracing::trace!("start_transaction");
 	}
 
 	fn storage_rollback_transaction(&mut self) -> Result<(), ()> {
 		self.mark_dirty();
-		self.overlay.rollback_transaction().map_err(|_| ())
+		self.overlay.rollback_transaction().map_err(|_| ())?;
```
Comment (bkchr) [inline primitives/state-machine/src/ext.rs:630]: You should not early return here to get the trace log. Same applies for below.

**Comment by bkchr** [comment]
> Currently all traces in `state-machine` evaluate to noop when compiled to a WASM binary.

How is that even important here? `state-machine` is a node "thing" and never used from the WASM side for "normal" blockchains. (Parachains actually will use this code in Wasm for their validation, however that is not related to what you are doing here ;))

The general direction of using the `tracing` logging macros, is a good idea. Especially as they support more structured logging.

**Comment by gnunicorn** [comment]
any movement on this?

**Comment by emostov** [comment]
> any movement on this?

I think it may be dropped in favor of #7780.
@dvdplm and/or @insipx may have some insight.

**Comment by gnunicorn** [comment]
closing due to a lack of progress.

**Pull request closed without merging**